src/Desktop.Analyzers/Core/DoNotUseInsecureDTDProcessing.cs

```diff
@@ -367,475 +326,520 @@ private void AnalyzeNodeForDtdProcessingOverloads(SyntaxNodeAnalysisContext cont
-                else if (property.MatchPropertyDerivedByName(_xmlTypes.DataViewManager, SecurityMemberNames.DataViewSettingCollectionString))   //FxCop CA3065
+                else
                 {
-                    DiagnosticDescriptor rule = RuleDoNotUseInsecureDTDProcessing;
-                    context.ReportDiagnostic(
-                        Diagnostic.Create(
-                            rule,
-                            node.GetLocation(),
-                            SecurityDiagnosticHelpers.GetLocalizableResourceString(
-                                nameof(DesktopAnalyzersResources.ReviewDtdProcessingPropertiesMessage),
-                                SecurityDiagnosticHelpers.GetNonEmptyParentName(node, semanticModel)
-                            )
-                        )
-                    );
+                    _objectCreationOperationsAnalyzed.Add(objCreation);
+                }
+
+                if (SecurityDiagnosticHelpers.IsXmlDocumentCtorDerived(objCreation.Constructor, _xmlTypes))
+                {
+                    AnalyzeObjectCreationForXmlDocument(context, variable, objCreation);
+                }
+                else if (SecurityDiagnosticHelpers.IsXmlTextReaderCtorDerived(objCreation.Constructor, _xmlTypes))
+                {
+                    AnalyzeObjectCreationForXmlTextReader(context, variable, objCreation);
+                }
+                else if (SecurityDiagnosticHelpers.IsXmlReaderSettingsCtor(objCreation.Constructor, _xmlTypes))
+                {
+                    AnalyzeObjectCreationForXmlReaderSettings(context, variable, objCreation);
+                }
+                else
+                {
+                    AnalyzeMethodOverloads(context, objCreation.Constructor, objCreation.ArgumentsInParameterOrder, objCreation.Syntax.GetLocation());
                 }
             }
 
-            private void AnalyzeNodeForXmlDocument(SyntaxNodeAnalysisContext context)
+            private void AnalyzeObjectCreationForXmlDocument(OperationAnalysisContext context, ISymbol variable, IObjectCreationExpression objCreation)
             {
-                SyntaxNode node = context.Node;
-                SemanticModel model = context.SemanticModel;
-
-                node = _syntaxNodeHelper.GetVariableDeclaratorOfAFieldDeclarationNode(node) ?? node;
-
-                SyntaxNode lhs = _syntaxNodeHelper.GetAssignmentLeftNode(node);
-                SyntaxNode rhs = _syntaxNodeHelper.GetAssignmentRightNode(node);
+                XmlDocumentEnvironment env = null;
 
-                if (lhs == null || rhs == null)
+                if (variable == null || !_xmlDocumentEnvironments.ContainsKey(variable))
                 {
-                    return;
+                    env = new XmlDocumentEnvironment();
+                    env.IsSecureResolver = false;
+                    env.IsXmlResolverSet = false;
+                    env.XmlDocumentDefinition = objCreation.Syntax;
```

Comment:
`AnalyzeObjectCreationForXmlDocument` is the same in both `if` and `else` case. Can move this statement out